File: app/src/main/java/com/ruzhan/recyclerviewitemanimation/holder/TwoSlideViewHolder.java
package com.ruzhan.recyclerviewitemanimation.holder;

import android.graphics.Color;
import android.view.View;
import android.widget.TextView;

import com.ruzhan.recyclerviewitemanimation.R;

import zhan.library.slide.holder.SlideViewHolder;

/**
 * Created by zhan on 2017/2/3.
 */

public class TwoSlideViewHolder extends SlideViewHolder {

    private View itemRoot;
    private View mContentLl;
    private TextView itemRedTv;
    private TextView itemBlueTv;
    private TextView helloTv;

    private static int mCurrentColor;
    private float mFraction;

    public TwoSlideViewHolder(View itemView) {
        super(itemView);

        itemRoot = itemView.findViewById(R.id.item_root);
        mContentLl = itemView.findViewById(R.id.content_ll);
        itemRedTv = (TextView) itemView.findViewById(R.id.item_red_tv);
        itemBlueTv = (TextView) itemView.findViewById(R.id.item_blue_tv);
        helloTv = (TextView) itemView.findViewById(R.id.hello_tv);
    }

    @Override
    public void doAnimationSet(int offset, float fraction) {
        mContentLl.scrollTo(-offset, 0);

        mFraction = 1 - fraction;

        itemRedTv.setScaleX(mFraction);
        itemRedTv.setScaleY(mFraction);

        itemBlueTv.setScaleX(mFraction);
        itemBlueTv.setScaleY(mFraction);

        helloTv.setScaleX(mFraction);
        helloTv.setScaleY(mFraction);
        helloTv.setAlpha(mFraction * 255);

        float blue = 233 * mFraction;
        mCurrentColor = Color.argb(233, 233 / 2, 233, (int) blue);
        itemRoot.setBackgroundColor(mCurrentColor);
    }

    @Override
    public void onBindSlideClose(int state) {
        itemRoot.setBackgroundColor(mCurrentColor);

    }

    @Override
    public void doAnimationSetOpen(int state) {
        itemRoot.setBackgroundColor(mCurrentColor);

        helloTv.setScaleX(mFraction);
        helloTv.setScaleY(mFraction);
        helloTv.setAlpha(mFraction * 255);
    }

    public void bind() {
        setOffset(150);

        //slide must call
        onBindSlide(mContentLl);
    }
}

File: app/src/main/java/com/ruzhan/recyclerviewitemanimation/holder/TwoSlideViewHolder.kt
package com.ruzhan.recyclerviewitemanimation.holder

import android.graphics.Color
import android.view.View
import kotlinx.android.synthetic.main.two_item.view.*
import zhan.library.slide.holder.SlideViewHolder

class TwoSlideViewHolder(itemView: View) : SlideViewHolder(itemView) {

    companion object {
        private var mCurrentColor = 0
        private const val DEFAULT_OFFSET = 150
    }

    private var mFraction: Float = 0.toFloat()

    override fun doAnimationSet(offset: Int, fraction: Float) {
        itemView.content_ll.scrollTo(-offset, 0)
        mFraction = 1 - fraction

        itemView.item_red_tv.scaleX = mFraction
        itemView.item_red_tv.scaleY = mFraction

        itemView.item_blue_tv.scaleX = mFraction
        itemView.item_blue_tv.scaleY = mFraction

        itemView.hello_tv.scaleX = mFraction
        itemView.hello_tv.scaleY = mFraction
        itemView.hello_tv.alpha = mFraction * 255

        val blue = 233 * mFraction
        mCurrentColor = Color.argb(233, 233 / 2, 233, blue.toInt())
        itemView.item_root.setBackgroundColor(mCurrentColor)
    }

    override fun onBindSlideClose(state: Int) {
        itemView.item_root.setBackgroundColor(mCurrentColor)
    }

    override fun doAnimationSetOpen(state: Int) {
        itemView.item_root.setBackgroundColor(mCurrentColor)

        itemView.hello_tv.scaleX = mFraction
        itemView.hello_tv.scaleY = mFraction
        itemView.hello_tv.alpha = mFraction * 255
    }

    fun bind() {
        offset = DEFAULT_OFFSET
        //slide must call
        onBindSlide(itemView.content_ll)
    }
}
